### ISSUE TITLE
[Menu] Group heading text inside menu does not have sufficient contrast in Dark Mode

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3024]
### Figma
In Figma group heading for menu in Sanoma Learning Dark Mode has contrast ratio of 2.2:1 between text color (#5D5D5D) and background color (#25292F).

<img width="458" height="441" alt="Image" src="https://github.com/user-attachments/assets/ba9ffac7-738a-45a3-96a1-cc5da0c0ea89" />

### Storybook
In [Menu Example](https://storybook.sanomalearning.design/?path=/story/actions-menu--example&globals=mode:dark) and [Menu Group With Heading](https://storybook.sanomalearning.design/?path=/story/actions-menu-menu--group-with-heading&globals=mode:dark) group heading for menu in Sanoma Learning Dark Mode has contrast ratio of 4.4:1 between text color (#8B8E91) and background color (#252930).

<img width="252" height="323" alt="Image" src="https://github.com/user-attachments/assets/816a5e02-7024-4073-9523-ff01f72b1bb6" />

## Proposed solution
For Figma and Storybook contrast ratio between heading group text and background should be equal to or greater than 4.5:1. 
This means changing heading group text color to eg. #8E9194 should fix this issue for background color used in Figma and Storybook Sanoma Learning Dark Mode.

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS team
